src/Services/SearchService.php

```diff
@@ -98,17 +105,83 @@ public function getResults()
         return $this->results;
     }
 
+    /**
+     * @param Response $results
+     * @param ExternalSearch $externalSearch
+     */
+    public function doSearch(Response $results, ExternalSearch $externalSearch) {
+        $productsIds = $this->filterInvalidVariationIds($results->getVariationIds());
+
+        /** @var ExternalSearch $searchQuery */
+        $externalSearch->setResults($productsIds, $results->getResultsCount());
+    }
+
+    /**
+     * @param HttpRequest $request
+     * @param ExternalSearch $externalSearch
+     */
+    public function doNavigation(HttpRequest $request, ExternalSearch $externalSearch) {
+        $searchResults = $this->fallbackSearchService->handleSearchQuery($request, $externalSearch);
+
+        $getIdsFromSearchResultItemsDocuments = function ($document) {
```

Comment:
* Move all mapping logic into `FallbackSearchService::handleSearchQuery`.
* The mentioned method should  create a response object e.g. `$this->responseParser->createResponseObject()`, process the results from Plenty and store the data in the created response object.
* Return the response object.

The general idea is that you don't have to parse any data in this class, since that's explicit knowledge that should only be present in `FallbackSearchService`.